src/cdk-experimental/menu/context-menu.ts

```diff
@@ -0,0 +1,318 @@
+  ],
+})
+export class CdkContextMenuTrigger implements OnDestroy {
+  /** Template reference variable to the menu to open on right click. */
+  @Input('cdkContextMenuTriggerFor')
+  get menuPanel(): CdkMenuPanel {
+    return this._menuPanel;
+  }
+  set menuPanel(panel: CdkMenuPanel) {
+    this._menuPanel = panel;
+
+    if (this._menuPanel) {
+      this._menuPanel._menuStack = this._menuStack;
+    }
+  }
+  /** Reference to the MenuPanel this trigger toggles. */
+  private _menuPanel: CdkMenuPanel;
+
+  /** Emits when the attached menu is requested to open. */
+  @Output('cdkContextMenuOpened') readonly opened: EventEmitter<void> = new EventEmitter();
+
+  /** Emits when the attached menu is requested to close. */
+  @Output('cdkContextMenuClosed') readonly closed: EventEmitter<void> = new EventEmitter();
+
+  /** Whether the context menu should be disabled. */
+  @Input('cdkContextMenuDisabled')
+  get disabled() {
+    return this._disabled;
+  }
+  set disabled(value: boolean) {
+    this._disabled = coerceBooleanProperty(value);
+  }
+  private _disabled = false;
+
+  /** A reference to the overlay which manages the triggered menu. */
+  private _overlayRef: OverlayRef | null = null;
+
+  /** The content of the menu panel opened by this trigger. */
+  private _panelContent: TemplatePortal;
+
+  /** Emits when the element is destroyed. */
+  private readonly _destroyed: Subject<void> = new Subject();
+
+  /** Reference to the document. */
+  private readonly _document: Document;
+
+  /** Emits when the document listener should stop. */
+  private readonly _stopDocumentListener = merge(this.closed, this._destroyed);
+
+  /** The menu stack for this trigger and its associated menus. */
+  private readonly _menuStack = new MenuStack();
+
+  constructor(
+    protected readonly _viewContainerRef: ViewContainerRef,
+    private readonly _overlay: Overlay,
+    private readonly _contextMenuTracker: ContextMenuTracker,
+    @Inject(CDK_CONTEXT_MENU_OPTIONS) private readonly _options: ContextMenuOptions,
+    @Inject(DOCUMENT) document: any,
+    @Optional() private readonly _directionality?: Directionality
+  ) {
+    this._document = document;
+
+    this._setMenuStackListener();
+  }
+
+  /**
+   * Open the attached menu at the specified location.
+   * @param coordinates where to open the context menu
+   */
+  openMenu(coordinates: ContextMenuCoordinates) {
+    if (this.disabled) {
+      return;
+    } else if (this.isMenuOpen()) {
+      // since we're moving this menu we need to close any submenus first otherwise they end up
+      // disconnected from this one.
+      this._menuStack.closeSubMenuOf(this._menuPanel._menu!);
+      this._overlayRef!.updatePositionStrategy(this._getOverlayPositionStrategy(coordinates));
+    } else {
+      this.opened.next();
+
+      if (this._overlayRef) {
+        this._overlayRef!.updatePositionStrategy(this._getOverlayPositionStrategy(coordinates));
```

Comment:
Instead of recreating the entire position strategy on each open, you should be able to only update the coordinates through `FlexibleConnectedPositionStrategy.setOrigin`.